File: python/examples/lux3d_img_to_3d.py
#!/usr/bin/env python3
"""
Minimal example: image-to-3D with Lux3D, poll until model URLs are ready.

v2.0-preview (default): returns .zip, .glb and .usdz formats
v1.0-pro:               returns a single .lux3d format

Usage: AHOLO_API_KEY=xxx python lux3d_img_to_3d.py ./chair.png
"""
import os
import sys
import time

from manycore.aholo_sdk_core import AholoClientConfig
from manycore.aholo_sdk_lux3d import create_lux3d_client

if len(sys.argv) < 2:
    raise SystemExit("Usage: AHOLO_API_KEY=xxx python lux3d_img_to_3d.py <image-path>")

file_path = sys.argv[1]
region = os.environ.get("AHOLO_REGION", "cn")
lux3d = create_lux3d_client(AholoClientConfig(region=region))

print(f"Creating img-to-3D task from {file_path} ...")
task_id = lux3d.img_to_3d.create_from_file(file_path)
print(f"taskId={task_id}, polling...")

t0 = time.monotonic()
result = lux3d.tasks.wait_for(task_id)
print(f"Task complete ({int((time.monotonic() - t0) * 1000)}ms)")
# v2.0-preview outputs: [zip, glb, usdz]
for output in result.get("outputs", []):
    print(output.get("content"))

File: java/examples/src/main/java/com/manycoreapis/sdk/examples/Lux3dImgTo3d.java
package com.manycoreapis.sdk.examples;

import com.manycoreapis.sdk.core.AholoClientConfig;
import com.manycoreapis.sdk.lux3d.Lux3dClient;
import com.manycoreapis.sdk.lux3d.model.TaskResult;

import java.nio.file.Paths;

/**
 * Minimal example: image-to-3D with Lux3D, poll until the model URLs are ready.
 *
 * v3.0-standard (default): zip + glb + optional usdz/obj/fbx (outputs[0..4])
 * v2.0-preview:              zip + glb + usdz
 * v1.0-pro:                  single .lux3d format
 *
 * Usage:
 *   AHOLO_API_KEY=xxx mvn exec:java -pl examples \
 *     -Dexec.mainClass=com.manycoreapis.sdk.examples.Lux3dImgTo3d \
 *     -Dexec.args="/path/to/chair.png"
 */
public class Lux3dImgTo3d {
    public static void main(String[] args) throws Exception {
        if (args.length < 1) {
            System.err.println("Usage: Lux3dImgTo3d <image-path>");
            System.exit(1);
        }

        String filePath = args[0];
        String region = System.getenv().getOrDefault("AHOLO_REGION", "cn");
        Lux3dClient lux3d = Lux3dClient.create(AholoClientConfig.ofRegion(region));

        System.out.println("Creating img-to-3D task from " + filePath + " ...");
        long taskId = lux3d.imgTo3d().createFromFile(Paths.get(filePath));
        System.out.println("taskId=" + taskId + ", polling...");

        long t0 = System.currentTimeMillis();
        TaskResult result = lux3d.tasks().waitFor(taskId);
        System.out.printf("Task complete (%dms)%n", System.currentTimeMillis() - t0);
        // v3.0-standard outputs: [zip, glb, usdz?, obj?, fbx?] — optional slots may be NOT_REQUESTED
        result.outputs().forEach(o -> o.content().ifPresent(System.out::println));
    }
}
